src/stores/search/searchReducer.ts

```diff
@@ -31,19 +31,38 @@ export function searchReducer(state = initialState, action: Redux.AnyAction): Se
   switch (action.type) {
     case SearchActions.SET_PENDING_OPTIONS: {
       const { sort, filter } = action.payload as SearchOptionsModel;
+      if (sort && !filter) {
```

Comment:
to avoid undefined and reduce the number of conditions you can try
```

const pendingOptions = {
      sort,
      filter
};

Object.entries(pendingOptions).forEach(([key, value]) => {
  if (!value) {
     delete pendingOptions[key]
  }
})


```